latte/models/cca/estimation.py

```diff
@@ -0,0 +1,74 @@
+import dataclasses
+
+import numpy as np
+import torch
+from sklearn.cross_decomposition import CCA
+
+from latte.tools.ksg import naive_ksg
+from latte.dataset import utils as dataset_utils
+
+
+@dataclasses.dataclass
+class CCAResult:
+    """A dataclass for holding the result of mutual information estimation using CCA
+
+    Members:
+        mutual_information (float): The estimate of the mutual information in the optimal as calculated on the
+                                    test dataset
+        mutual_information (float): The estimate of the mutual information in the complement of the optimal subspace
+                                    as calculated on the test dataset
+        A_1: In case of also estimating the linear subspace capturing the most information, this holds the
+           d-frame defining the estimated subspace of the first distribution
+        A_2: In case of also estimating the linear subspace capturing the most information, this holds the
+           d-frame defining the estimated subspace of the second distributions
+        E_1: In case of also estimating the linear subspace capturing the most information, this holds the
```

Comment:
Is it a d-frame spanning the subspace or a projection matrix? It's calculated as 
`np.eye(A_hat_1.shape[0]) - A_hat_1 @ A_hat_1.T`
so I think it's rather a projection matrix.

Reply:
Awesome catch! My notation may be a bit sloppy here, yeah...